src/mission_companion.cpp

```diff
@@ -466,6 +575,63 @@ bool talk_function::outpost_missions( npc &p, std::string id, std::string title
     if (cur_key == "Recover Ally from Menial Labor"){
         labor_return(p);
     }
+
+    if (cur_key == "Upgrade Camp"){
+        const recipe *making = &recipe_id( bldg ).obj();
+         //Stop upgrade if you don't have materials
+        if( making->requirements().can_make_with_inventory( total_inv, 1 ) ){
+            std::vector<std::shared_ptr<npc>> npc_list = companion_list( p, "_faction_upgrade_camp" );
+            if (npc_list.size() < 1) {
```

Comment:
This is just cryptic. Check for the list being empty instead. There is a specific function for that.